### PR TITLE
refactor(package): update `webpack-defaults` v1.6.0...2.6.0 (`devDependencies`)

### PR DESCRIPTION
**Pull request by piecyk**
* update webpack defaults
* lint `./src`
* convert example as simple test 

### Type 

- [x] Refactor

### Issues 

- None

### SemVer

- [x] Major

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,72 +2,86 @@
...
+  "author": "Tobias Koppers @sokra",
+  "homepage": "https://github.com/webpack-contrib/thread-loader",
+  "bugs": "https://github.com/webpack-contrib/thread-loader/issues",
+  "bin": "",
   "main": "dist/cjs.js",
-  "files": [
-    "dist"
-  ],
+  "engines": {
+    "node": ">= 6.9.0 <7.0.0 || >= 8.9.0"
```
Comment (michael-ciniawsky) [inline package.json]: Please relax this to `"node": ">= 6"`

File: test/TestMemoryFS.test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import path from 'path';
```
Comment (michael-ciniawsky) [inline test/TestMemoryFS.test.js]: Please add test via a separate PR, looks good and is appreciated but we normally do it a bit different

File: package.json [hunk 3]
```diff
@@ -2,72 +2,86 @@
...
+    "ci:lint": "npm run lint && npm run security",
+    "ci:test": "npm run test -- --runInBand",
+    "ci:coverage": "npm run test:coverage -- --runInBand",
+    "defaults": "webpack-defaults"
+  },
+  "files": [
+    "dist/"
+  ],
+  "peerDependencies": {
+    "webpack": "^4.3.0"
```
[the remark this answers is not in the document]
Comment (piecyk) [reply package.json]: It should work, this was set by `defaults` so I assumed it's the correct way... 
Will change it to `"webpack": "^3.0.0 || ^4.0.0"`

File: package.json (hunk 1 above)
Comment (piecyk) [reply package.json]: 👍

File: test/TestMemoryFS.test.js (hunk 2 above)
Comment (piecyk) [reply test/TestMemoryFS.test.js]: Yep, was thinking the same 👍 Did it here as wanted to refactor the `./examples` to align overall structure... Will drop it from here

**Comment by piecyk** [comment]
ping @michael-ciniawsky

**Comment by piecyk** [comment]
Closing this, as it looks it will be not merged.

**Pull request closed without merging**